atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientImpl.java

```diff
@@ -277,6 +278,17 @@ public CqlResult execute_cql3_query(CqlQuery cqlQuery,
         return executeHandlingExceptions(() -> client.execute_cql3_query(queryBytes, compression, consistency));
     }
 
+    @Override
+    public void close() {
+        TProtocol inputProtocol = getInputProtocol();
+        TProtocol outputProtocol = getOutputProtocol();
+        try (TTransport inputTransport = inputProtocol.getTransport();
+                TTransport outputTransport = outputProtocol.getTransport()) {
+            inputProtocol.reset();
+            outputProtocol.reset();
```

Comment:
Verified `reset()` shouldn't throw, so it should be safe. Theoretically, if the first `reset()` call throws the second is never invoked and we might want to avoid that, but this is safe here.

Reply:
yeah, the `reset()` isn't actually necessary, mainly just wanted something for inside the try-with-resources, though could just put a `log.trace("Closing CassandraClient");` in the block.

Also debated explicitly clearing the `TFramedTransport` which holds a potentially large (~16MB) `readBuffer`. Don't have strong feelings either way, and this should be unnecessary as these clients with their referenced protocol, sockets, and buffers should be eligible for GC shortly after this point.

```
    @Override
    public void close() {
        TProtocol inputProtocol = getInputProtocol();
        TProtocol outputProtocol = getOutputProtocol();
        try (TTransport inputTransport = inputProtocol.getTransport();
                TTransport outputTransport = outputProtocol.getTransport()) {
            log.trace("Closing CassandraClient");
            tryClear(inputTransport);
            tryClear(outputTransport);
        }
    }

    private static void tryClear(TTransport transport) {
        if (transport instanceof TFramedTransport) {
            ((TFramedTransport) transport).clear();
        }
    }
```

We currently clear that buffer via reflection before returning valid clients to the pool in https://github.com/palantir/atlasdb/blob/94e7c6087896fb071bccd95b5bdaa320908e1403/atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolingContainer.java#L173-L194 ) and note that https://github.com/palantir/atlasdb/pull/2971 specifically sets an empty array in the `TMemoryInputTransport`, but that would not be an issue here on eviction.

Reply:
Yep, makes sense - thanks for the additional context!